### PR TITLE
Add analysis module permissions

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Fixes https://zube.io/deep/deep/c/3779

## Changes

* Add permissions in the analysis module
    * Users who cannot edit entries should not be able to view edit buttons or edit page

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/newViews/AnalysisModule/index.tsx [hunk 1]
```diff
@@ -115,6 +116,18 @@ const timelineTickSelector = (d: number) => {
...
+    hasAnalysisFramework,
+    entryPermissions,
+}: {
+    hasAnalysisFramework: boolean;
+    entryPermissions: {
+        create: boolean;
+        modify: boolean;
+    };
+}) {
+    return (!hasAnalysisFramework || !(entryPermissions.create || entryPermissions.modify));
```
Comment (tnagorra) [inline src/newViews/AnalysisModule/index.tsx:129]: Why not just use create?

File: src/newViews/AnalysisModule/Analysis/index.tsx [hunk 2]
```diff
@@ -76,6 +77,20 @@ const barChartMargin = {
...
+    hasAnalysisFramework,
+    entryPermissions,
+}: {
+    hasAnalysisFramework: boolean;
+    entryPermissions: {
+        create: boolean;
+        modify: boolean;
+    };
+}) {
+    return (!hasAnalysisFramework || !(entryPermissions.create || entryPermissions.modify));
```
Comment (tnagorra) [inline src/newViews/AnalysisModule/Analysis/index.tsx:90]: Why not just use modify?

File: src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx [hunk 3]
```diff
@@ -28,6 +29,19 @@ import { calcPercent } from '#utils/safeCommon';
...
+    hasAnalysisFramework,
+    entryPermissions,
+}: {
+    hasAnalysisFramework: boolean;
+    entryPermissions: {
+        create: boolean;
+        modify: boolean;
+    };
+}) {
+    return (!hasAnalysisFramework || !(entryPermissions.create || entryPermissions.modify));
```
Comment (tnagorra) [inline src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx:42]: Why not just use modify?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/newViews/AnalysisModule/index.tsx (hunk 1 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/index.tsx:129]: We've duplicated the edit entries permissions here

File: src/newViews/AnalysisModule/Analysis/index.tsx (hunk 2 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/index.tsx:90]: We've duplicated the edit entries permissions here

File: src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx (hunk 3 above)
Comment (AdityaKhatri) [reply src/newViews/AnalysisModule/Analysis/AnalysisPillar/index.tsx:42]: We've duplicated the edit entries permissions here

**Pull request merged**